File: app/src/main/java/de/karzek/diettracker/presentation/search/grocery/adapter/viewHolder/GrocerySearchFoodResultViewHolder.java
package de.karzek.diettracker.presentation.search.grocery.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.model.GroceryDisplayModel;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class GrocerySearchFoodResultViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.grocery_name)
    TextView groceryName;

    private final OnFoodSearchResultItemClickedListener onItemClickedListener;

    public GrocerySearchFoodResultViewHolder(ViewGroup viewGroup, OnFoodSearchResultItemClickedListener onItemClickedListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_grocery_search_food, viewGroup, false));
        ButterKnife.bind(this, itemView);
        this.onItemClickedListener = onItemClickedListener;
    }

    public void bind(GroceryDisplayModel foodSearchResultItem) {
        groceryName.setText(foodSearchResultItem.getName());
        itemView.setTag(foodSearchResultItem.getId());
    }

    @OnClick(R.id.grocery_search_food_item)
    public void onItemClicked() {
        onItemClickedListener.onItemClicked((int) itemView.getTag());
    }

    public interface OnFoodSearchResultItemClickedListener {
        void onItemClicked(int id);
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/search/grocery/adapter/viewHolder/GrocerySearchFoodResultViewHolder.kt
package de.karzek.diettracker.presentation.search.grocery.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.model.GroceryDisplayModel
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_grocery_search_food.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class GrocerySearchFoodResultViewHolder(override val containerView: View,
                                        private val onItemClickedListener: OnFoodSearchResultItemClickedListener
) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(foodSearchResultItem: GroceryDisplayModel) {
        grocery_name.text = foodSearchResultItem.name
        itemView.tag = foodSearchResultItem.id

        grocery_search_food_item.setOnClickListener { onItemClickedListener.onItemClicked(itemView.tag as Int) }
    }

    interface OnFoodSearchResultItemClickedListener {
        fun onItemClicked(id: Int)
    }

}
